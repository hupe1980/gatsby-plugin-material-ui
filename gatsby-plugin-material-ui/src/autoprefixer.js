import postcss from "postcss";
import autoprefixer from "autoprefixer";

export default function(css, pathname) {
  const prefixer = postcss([autoprefixer]);

  try {
    return prefixer.process(css, { from: undefined }).css;
  } catch (error) {
    if (error.name === `CssSyntaxError`) {
      throw new Error(`Pathname: ${pathname} ${error.toString()}`);
    }
    throw error;
  }
}
