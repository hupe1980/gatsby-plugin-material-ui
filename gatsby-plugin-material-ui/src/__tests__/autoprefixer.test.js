import autoprefixer from "../autoprefixer";

describe(`autoprefixer`, () => {
  describe(`default`, () => {
    const invalidCss = `
        .jss1 {
            border-radius: 6px;
        }
        .jss1 {
            h1, h2, h3: [object Object];
        }
    `;

    it(`should throw an error if css is invalid1`, () => {
      expect(() =>
        autoprefixer(invalidCss, `/test`),
      ).toThrowErrorMatchingSnapshot();
    });
  });
});
