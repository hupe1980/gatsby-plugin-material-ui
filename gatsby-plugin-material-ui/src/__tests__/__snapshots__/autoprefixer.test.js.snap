// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`autoprefixer default should throw an error if css is invalid1 1`] = `
"Pathname: /test CssSyntaxError: <css input>:6:17: Unknown word

  4 |         }
  5 |         .jss1 {
> 6 |             h1, h2, h3: [object Object];
    |                 ^
  7 |         }
  8 |     
"
`;
